.github/workflows/main.yml

```diff
@@ -0,0 +1,104 @@
+on:
+  push:
+    branches:
+      - master
+      - stable
+  pull_request:
+    branches:
+      - master
+name: Linux CI
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Linux build & test
+    env:
+      PROCESSORS: 0
+      PREFIX: "/tmp/ADALIB_DIR"
+      GPR_PROJECT_PATH: /tmp/ADALIB_DIR/share/gpr
+    steps:
+      - name: Get langkit
+        uses: actions/checkout@v2
+
+      - name: Setup Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+
+      - name: Install python dependencies
+        run: |
+            pip install -r REQUIREMENTS.dev
+            pip install jsonschema
+            pip install mypy==0.780
+            python setup.py install
+            pip install flake8
+
+      - name: Get gprbuild for libgpr
+        uses: actions/checkout@v2
+        with:
+          repository: AdaCore/gprbuild
+          path: gprbuild
+
+      - name: Get gnatcoll core
+        uses: actions/checkout@v2
+        with:
+          repository: AdaCore/gnatcoll-core
+          path: gnatcoll-core
+
+      - name: Get gnatcoll iconv/gmp
+        uses: actions/checkout@v2
+        with:
+          repository: AdaCore/gnatcoll-bindings
+          path: gnatcoll-bindings
+
+      - uses: actions/cache@v2
+        with:
+          path: ./cached_gnat
+          key: ${{ runner.os }}-gnat-ce-2020
+      - name: Get GNAT Community 2020 toolchain
+        uses: ada-actions/toolchain@3f69585
+        with:
+          distrib: community
+          install_dir: ./cached_gnat
+
+      # This step caches just the obj directories, so that when nothing has
+      # changed in the dependencies, the build is much faster.
+      - name: Cache build artifacts
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: |
+              gprbuild/gpr/libobj
+              gnatcoll-core/obj
+              gnatcoll-bindings/gmp/obj
+              gnatcoll-bindings/iconv/obj
+              contrib/python/build/obj
+              contrib/lkt/build/obj
+          key: ${{ runner.os }}-build-artifacts-2
```

Comment:
Just curious: why the `-2`?

Reply:
> Also, more curious: gprbuild will rebuild sources if source files have more recent modification time than build artifacts. Are git checkouts cached as well? Otherwise the cache will not be used (git clone = more recent modification times).

That's not what I've seen in practice. Aren't creation and modification times different ?

Reply:
> You have no way of invalidating cache entries yourself. 

That sounds scary…

> That's not what I've seen in practice.

Ok, it’s not blocking anyway (at worse, we’ll just have inefficient builds, I guess)

> Aren't creation and modification times different ? 
No, they are the same. IIUC, the other times are: access time (when file is read) and changed timestamp (file metadata is updated).

Reply:
> That sounds scary…

I don't know. It just means the UI for invalidating is not user friendly (you have to change the cache key).

> Ok, it’s not blocking anyway (at worse, we’ll just have inefficient builds, I guess)

I'll look at this at a later stage if you don't mind